client/src/AppBundle/Service/Audit/AwsAuditLogHandler.php

```diff
@@ -0,0 +1,174 @@
+<?php
+
+namespace AppBundle\Service\Audit;
+
+use Aws\CloudWatchLogs\CloudWatchLogsClient;
+use Aws\CloudWatchLogs\Exception\CloudWatchLogsException;
+use Monolog\Logger;
+
+class AwsAuditLogHandler extends AbstractAuditLogHandler
+{
+    /** @var CloudWatchLogsClient */
+    private $client;
+
+    /** @var string */
+    private $group;
+
+    /** @var string */
+    private $stream;
+
+    /** @var string */
+    private $sequenceToken;
+
+    /** @var array */
+    private $existingStreams = [];
+
+    /** @var bool */
+    private $initialized = false;
+
+    /**
+     * @param CloudWatchLogsClient $client
+     * @param $group
+     * @param int $level
+     * @param bool $bubble
+     */
+    public function __construct(CloudWatchLogsClient $client, $group, $level = Logger::NOTICE, $bubble = true)
+    {
+        $this->client = $client;
+        $this->group = $group;
+
+        parent::__construct($level, $bubble);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function write(array $entry): void
+    {
+        if (!$this->shallHandle($entry)) {
+            return;
+        }
+
+        $this->stream = $entry['context']['event'];
+        $entry = $this->formatEntry($entry);
+
+        if (false === $this->initialized) {
+            $this->initialize();
+        }
+
+        // send items, retry once with a fresh sequence token
+        try {
+            $this->send($entry);
+        } catch (CloudWatchLogsException $e) {
```

Comment:
Is this to account for expired tokens?

Reply:
This try catch was borrowed from the lib mentioned in the description, but yes, from what I can tell it allows for another attempt with a fresh sequence token.